dts/arm/nordic/nrf91ns.dtsi

```diff
@@ -6,6 +6,7 @@
 
 #include <arm/armv8-m.dtsi>
 #include <nordic/nrf_common.dtsi>
+#include <zephyr/dt-bindings/adc/nrf-saadc-nrf5x_v2.h>
```

Comment:
note: confusing as this is nrf**91**?

Reply:
also:
> I wonder if binding headers should just be more specific (ie, nrf91, nrf52833, etc.) instead of having _v suffix. It looks like this is the same IP just with more/less channels routed?

the problem is that nRF devices share available SAADC internal channel inputs in non-family-unified scheme :
- AINx -> available on nRF52/53/91/54H/NRF54L -> `nrf-saadc-nrf5x`
- VDD -> available on nRF52/53/91/NRF54L (no nRF54H)  -> `nrf-saadc-nrf5x_v2`
- VDDHDIV5 -> available on nRF52833/nRF52840/nRF5340  -> `nrf-saadc-nrf5x_v3`
- DVDD & AVDD -> available on nRF54L only  -> `nrf-saadc-nrf54lx`

My attempt was to group the devices by available functionality. I could do it per-device, but I guess this might lead to some duplication

Reply:
> * AINx -> available on nRF52/53/91/54H/NRF54L -> `nrf-saadc-nrf5x`

`-nrf5x` is obviously wrong if it is also to be used for nrf91. Why not just `nrf-saadc`?

> * VDD -> available on nRF52/53/91/NRF54L (no nRF54H)  -> `nrf-saadc-nrf5x_v2`
> * VDDHDIV5 -> available on nRF52833/nRF52840/nRF5340  -> `nrf-saadc-nrf5x_v3`
> * DVDD & AVDD -> available on nRF54L only  -> `nrf-saadc-nrf54lx`

How about just `nrf-saadc-v2`, `nrf-saadc-v3`, and `nrf-saadc-v4`?